Release tarball format

# Required information

 * Distribution: Macaroni OS, Funtoo, Gentoo

# Issue description

Hi, the GitHub Release tarball doesn't contain the libraries raft and dqlite used in the Gentoo and Funtoo ebuilds without the need to have separate packages for these dependencies.

Are there plans to fix this and supply a tarball like was available on the linuxcontainer.org site?

Thanks in advance

Yeah, sorry, The vendors are present on the release tarball but there is something changed. I'm receiving issue on linking phase about raft:
```
# github.com/canonical/lxd/lxd-migrate
/usr/lib/go/pkg/tool/linux_amd64/link: running x86_64-pc-linux-gnu-gcc failed: exit status 1
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_calloc'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_add'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_state'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_heartbeat_timeout'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_last_index'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_errmsg'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_free'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_apply'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_pre_vote'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_configuration_add'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_remove'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_bootstrap'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_recover'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_close'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_init'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_uv_set_block_size'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_configuration_init'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_leader'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_uv_close'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_max_catch_up_rounds'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_max_catch_up_round_duration'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_configuration_close'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_digest'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_snapshot_threshold'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_malloc'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_assign'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_barrier'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_uv_set_auto_recovery'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_last_applied'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_snapshot_trailing'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_start'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_strerror'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_set_election_timeout'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_uv_init'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/app-emulation/lxd-5.17/work/lxd-5.17/vendor/dqlite/.libs//libdqlite.so: undefined reference to `raft_transfer'
collect2: error: ld returned 1 exit status
```

Just to share a reference to the ebuild used: I just renamed this: https://github.com/geaaru/geaaru_overlay/blob/funtoo/app-emulation/lxd/lxd-5.16.ebuild in 5.17.

I'm investigating... thanks for the fast feedback

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -12,6 +12,11 @@ concurrency:
 
 jobs:
   code-tests:
+    env:
+      CGO_CFLAGS: "-I/home/runner/work/lxd/lxd-test/vendor/raft/include/ -I/home/runner/work/lxd/lxd-test/vendor/dqlite/include/"
+      CGO_LDFLAGS: "-L/home/runner/work/lxd/lxd-test/vendor/raft/.libs/ -L/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+      LD_LIBRARY_PATH: "/home/runner/work/lxd/lxd-test/vendor/raft/.libs/:/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+      CGO_LDFLAGS_ALLOW: "(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
     name: Code tests
     runs-on: ubuntu-22.04
     steps:
@@ -30,17 +35,16 @@ jobs:
       - name: Install dependencies
         run: |
           sudo add-apt-repository ppa:ubuntu-lxc/lxc-git-master -y --no-update
-          sudo add-apt-repository ppa:dqlite/dev -y --no-update
           sudo apt-get update
 
           sudo apt-get install --no-install-recommends -y \
+            build-essential \
             curl \
             gettext \
             git \
             libacl1-dev \
             libcap-dev \
             libdbus-1-dev \
-            libdqlite-dev \
             liblxc-dev \
             lxc-templates \
             libseccomp-dev \
@@ -48,6 +52,7 @@ jobs:
             libsqlite3-dev \
             libtool \
             libudev-dev \
+            libuv1-dev \
             make \
             pkg-config \
             shellcheck
@@ -57,6 +62,19 @@ jobs:
       - name: Download go dependencies
         run: |
           go mod download
+
+      - name: Make LXD tarball and unpack it
+        env:
+          CUSTOM_VERSION: "test"
+        run: |
+          make dist
+          tar -xzf lxd-test.tar.gz -C ~/work/lxd/
+          rm lxd-test.tar.gz
+
+      - name: Build LXD dependencies
+        run: |
+          cd ~/work/lxd/lxd-test
+          make deps
 
       - name: Run LXD build
         run: |
@@ -68,7 +86,7 @@ jobs:
 
       - name: Unit tests (all)
         run: |
-          sudo go test ./...
+          sudo --preserve-env=CGO_CFLAGS,CGO_LDFLAGS,CGO_LDFLAGS_ALLOW,LD_LIBRARY_PATH LD_LIBRARY_PATH=${LD_LIBRARY_PATH} go test ./...
 
   system-tests:
     env:

--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ POFILES=$(wildcard po/*.po)
 MOFILES=$(patsubst %.po,%.mo,$(POFILES))
 LINGUAS=$(basename $(POFILES))
 POTFILE=po/$(DOMAIN).pot
-VERSION=$(shell grep "var Version" shared/version/flex.go | cut -d'"' -f2)
+VERSION=$(or ${CUSTOM_VERSION},$(shell grep "var Version" shared/version/flex.go | cut -d'"' -f2))
 ARCHIVE=lxd-$(VERSION).tar
 HASH := \#
 TAG_SQLITE3=$(shell printf "$(HASH)include <dqlite.h>\nvoid main(){dqlite_node_id n = 1;}" | $(CC) ${CGO_CFLAGS} -o /dev/null -xc - >/dev/null 2>&1 && echo "libsqlite3")

